gateway/ln-gateway/src/lib.rs

```diff
@@ -842,13 +850,24 @@ impl Gateway {
         }: SetConfigurationPayload,
     ) -> Result<()> {
         let gw_state = self.state.read().await.clone();
-        let can_set_config = matches!(gw_state, GatewayState::Running { .. })
-            || matches!(gw_state, GatewayState::Configuring)
-            || matches!(gw_state, GatewayState::Connected);
-        if !can_set_config {
+        if matches!(gw_state, GatewayState::Disconnected) {
             return Err(GatewayError::Disconnected);
         }
 
+        let lightning_network = match gw_state {
+            GatewayState::Running {
+                lightning_network, ..
+            } => {
+                if network.is_some() && network != Some(lightning_network) {
+                    return Err(GatewayError::GatewayConfigurationError(
+                        "Cannot change network while connected to a lightning node".to_string(),
+                    ));
+                }
+                lightning_network
+            }
+            _ => DEFAULT_NETWORK,
```

Comment:
Let's add a comment here that explains that, if the gateway is not in the `Running` state, the network will be set later once the connection to the lightning node is established.

Reply:
💯